handler/src/main/java/io/netty/handler/ssl/SslHandler.java

```diff
@@ -1062,7 +1062,15 @@ private SSLEngineResult wrap(ByteBufAllocator alloc, SSLEngine engine, ByteBuf i
     public void channelInactive(ChannelHandlerContext ctx) throws Exception {
         boolean handshakeFailed = handshakePromise.cause() != null;
 
+        // Channel closed, we will generate 'ClosedChannelException' now.
         ClosedChannelException exception = new ClosedChannelException();
+
+        // Add a supressed exception if the handshake was not completed yet.
+        if (!isStateSet(STATE_HANDSHAKE_STARTED) || handshakePromise.isDone()) {
+            exception.addSuppressed(new StacklessSSLHandshakeException("Connection closed before " +
```

Comment:
"before" doesn't help to understand if it was "before handshake started" or "after it started but before it complete". Consider clarifying that the handshake started, but didn't complete